src/java/org/apache/cassandra/tools/BulkLoadConnectionFactory.java

```diff
@@ -70,4 +70,9 @@ private NettyStreamingChannel create(OutboundConnectionSettings template, int me
 
```

Comment:
If a `connectTo` is always set in this last `create()` overload, the comment in the second `create()` is incorrect, given the preferred address is always overridden. From an API perspective, the `preferred` address is completely ignored, and the second `create()` overload is useless, and we could just remove it. (The `Factory` interface has a default method that ignores `preferred`.)

Reply:
...and just for the curious, I think we can't try to make this compatible w/ `EC2Snitch` either, given the whole point of this issue is that getting the preferred IP from the system keyspace is a no-no.